File: Chapter 03/C#/Section_3_3_5_DIP/Before/Forward.cs
using System;

namespace Chapter03.DIP.Before;

/// <summary>
/// A Low-Level Module.
/// 
/// ARCHITECTURE NOTE: This is a concrete implementation of a player. 
/// Because the Coach depends directly on this class, any changes to 
/// the Forward's method names or parameters will immediately break the 
/// Coach class. The dependency arrow points in the wrong direction.
/// </summary>
public class Forward
{
    public void Attack() => Console.WriteLine("  [Action] Forward is attacking.");
}

File: Chapter 03/JS/section_3_3_5_dip/before/forward.js
/**
 * A Low-Level Module.
 * * ARCHITECTURE NOTE: This is a concrete implementation. Because the Coach 
 * imports this file directly, any change to the 'attack' method name will 
 * force a breaking change in the Coach class.
 */
class Forward {
    attack() {
        console.log("  [Action] Forward is attacking.");
    }
}

module.exports = Forward;